src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -2014,6 +2037,120 @@ export default abstract class ComparisonStore {
         });
     }
 
+    readonly molecularProfilesInStudies = remoteData<MolecularProfile[]>(
+        {
+            await: () => [this.studies],
+            invoke: () => {
+                const studyIds = _.map(
+                    this.studies.result,
+                    (s: CancerStudy) => s.studyId
+                );
+                return client.fetchMolecularProfilesUsingPOST({
+                    molecularProfileFilter: {
+                        studyIds: studyIds,
+                    } as MolecularProfileFilter,
+                });
+            },
+        },
+        []
+    );
+
+    readonly customDriverAnnotationProfiles = remoteData<MolecularProfile[]>(
+        {
+            await: () => [this.molecularProfilesInStudies],
+            invoke: () => {
+                return Promise.resolve(
+                    _.filter(
+                        this.molecularProfilesInStudies.result,
+                        (molecularProfile: MolecularProfile) =>
+                            // discrete CNA's
+                            (molecularProfile.molecularAlterationType ===
+                                AlterationTypeConstants.COPY_NUMBER_ALTERATION &&
+                                molecularProfile.datatype ===
+                                    DataTypeConstants.DISCRETE) ||
+                            // mutations
+                            molecularProfile.molecularAlterationType ===
+                                AlterationTypeConstants.MUTATION_EXTENDED ||
+                            // structural variants
+                            molecularProfile.molecularAlterationType ===
+                                AlterationTypeConstants.STRUCTURAL_VARIANT
+                    )
+                );
+            },
+        },
+        []
+    );
+
+    readonly customDriverAnnotationReport = remoteData<IDriverAnnotationReport>(
+        {
+            await: () => [this.customDriverAnnotationProfiles],
+            invoke: async () => {
+                const molecularProfileIds = _.map(
+                    this.customDriverAnnotationProfiles.result,
+                    (p: MolecularProfile) => p.molecularProfileId
+                );
+                const report = await internalClient.fetchAlterationDriverAnnotationReportUsingPOST(
+                    {
+                        molecularProfileIds,
+                    }
+                );
+                return {
+                    ...report,
+                    hasCustomDriverAnnotations:
+                        report.hasBinary || report.tiers.length > 0,
+                };
+            },
+            onResult: result => {
+                initializeCustomDriverAnnotationSettings(
```

Comment:
will this/should this re-init the settings every time this invokes?

Reply:
Yes, I think it should. Since this method may be called multiple times after the cange of molecularProfilesInStudies to molecularProfilesInActiveStudie, would you prefer a rename to updateCustomDriverAnnotationSettings?